src/lib/player.coffee

```diff
@@ -78,4 +82,119 @@ class Player
 
         @time = newTime
 
-module.exports = Player
+    render: (ctx) ->
+        # Cast
+        if @startCastTime?
+            realCastTime = Utils.game.speedInverse(Skills[@castedSkill].castTime)
+            radiusMs = @radius / realCastTime
+            radius = (radiusMs * (@time - @startCastTime)) + @radius
+
+            angle = @p.angle @castP
+            halfCone = Skills[@castedSkill].cone / 2
+
+            ctx.beginPath()
+            ctx.moveTo @p
+            ctx.arc @p, radius, angle - halfCone, angle + halfCone
+            ctx.moveTo @p
+            ctx.fillStyle Skills[@castedSkill].color
+            ctx.fill()
+
+        # Location
+        ctx.filledCircle @p, @radius, @arena.teams[@team].color
+
+        # casting circle
+        if Config.UI.castingCircles
+            ctx.beginPath()
+            ctx.circle @p, @maxCastRadius
+            ctx.lineWidth 1
+            ctx.setLineDash [3,12]
+            ctx.strokeStyle "#777777"
+            ctx.stroke()
+            ctx.setLineDash []
+
+    clear: ->
+        super()
+
+class AIPlayer extends ProtoPlayer
+    constructor: (@arena, @handler, startP, team) ->
+        super @arena, startP, team
+
+    update: (newTime) ->
+        super newTime
+        otherPs = _.reject _.values(@handler.players), team: @team
+
+        if Math.random() < Utils.game.speed(0.005) and not @startCastTime?
+            @handler.fire @, _.sample(otherPs).p, 'orb'
+
+        chanceToMove = Math.random() < Utils.game.speed(0.03)
+        if not @startCastTime? and (chanceToMove or @p.equal @destP)
+            @handler.moveTo @, @arena.map.randomPoint()
+
+    moveTo: (@destP) ->
```

Comment:
these functions fall through automatically, don't think you need to redefine then super